rust/agents/relayer/src/msg/processor.rs

```diff
@@ -63,92 +66,106 @@ impl MessageProcessor {
         }
     }
 
+    /// Tries to get the next message to process.
+    ///
+    /// If no message with self.message_nonce is found, returns None.
+    /// If the message with self.message_nonce is found and has previously
+    /// been marked as processed, increments self.message_nonce and returns
+    /// None.
+    fn try_get_unprocessed_message(&mut self) -> Result<Option<HyperlaneMessage>, DbError> {
```

Comment:
Does this need to be `&mut self`? It looks like `&self` should be fine.

Reply:
We update self.message_nonce